docs/devel/consumers-claims.md

```diff
@@ -0,0 +1,297 @@
+A `LoadBalancerManager` is responsible for binding `IngressClaim`s to
+`LoadBalancers`s.  If the `IngressClaim` can not be satisified by any existing
+`LoadBalancer`, the `LoadBalancerManager` may provision a new one.
+
+```
+                       +--------------+
+        +--create----> | LoadBalancer |
+        |              +--------------+
+   +----+----+               ^
+   |  LBMgr  |---bind------> :
+   +----+----+               v
+        |              +---------------+
+        +---watch----> | IngressClaim  | <--+
+                       +---------------+    |
+                                 ^          |
+                                 |          +-------- user-created
+                                 |          |
+                           +----------+     |
+                           | Ingress  | <---+
+                           +----------+
+```
+
+### IngressClaim
+
+An `IngressClaim` does not specify which specific Ingress it
+wants.  Instead it specifies
+characteristics of the storage it needs and those characteristics are satisfied
+as best is possible.  Characteristics might include the availability properties
+required or the specific IP address to use.
+
+Like `PersistentVolumeClaim`, users will be able to request coarse "classes" of
+ingress (e.g. Gold, Silver, Bronze).  How those classes map to real storage
+technologies is a policy set by the cluster administrators.
+
+### LoadBalancerManagers
+
+The act of binding a `IngressClaim` to an `Ingress` is the responsibility of a
+process in the cluster.  A cluster may have any number of
+`LoadBalancerManager`s running, each of which handles some set of claim
+classes.  When an `IngressClaim` is created that asks for "Gold" ingress, the
+`LoadBalancerManager` which understands the "Gold" class will make the binding.
+
+`LoadBalancerManager`s can be simple `Pod`s running in the cluster, pulling
+cofiguration from hardcoded values, flags, config objects, or whatever
+mechanism makes sense for them.  We expect config objects to be a perfect fit
+for this use case.
+
+This very loose coupling means that an admin can define classes however they
+like.  For example, "Gold" might mean GCE Cloud Load Balancer, "Silver"
+might mean HAProxy, and "Bronze" might mean a port on a single VM.
+Adding a totally bespoke `LoadBalancerManager` is trivial.
+
+Note that classes are just identifiers - there are no special names or
+meanings.  One of the `LoadBalancerManager`s must be the "default" for claims that do
+not specify a class.
+
+### Config
+
+We expect that `LoadBalancerManager`s will need some amount of configuration, and it
+will be convenient if they have similar parameters, but this should be by
+convention, not by diktat.  Example parameters:
+   * Minimum free instances - how many ready-to-use balancers should be maintained
+   * Maximum free instances - limit of number of ready-to-use balancers
```

Comment:
Same question as before-- should this be limit on total number of balancers rather than unallocated (the idea being that if users are charged for balancers, they probably care about the total number regardless of whether they are in-use or ready-to-use).